int16 left
int16 right
int16 leftPos
int16 rightPos
